src/igvfd/audit/file_set.py

```diff
@@ -21,3 +21,128 @@ def audit_no_files(value, system):
             f'has no files.'
         )
         yield AuditFailure('missing files', detail, level='WARNING')
+
+
+@audit_checker('FileSet', frame='object')
+def audit_missing_seqspec(value, system):
+    '''
+        audit_detail: Sequence files in a file set are expected to link to a seqspec file.
+        audit_category: missing sequence specification
+        audit_levels: ERROR
+    '''
+    if 'files' in value:
+        no_seqspec = []
+        for file in value['files']:
+            if file.startswith('/sequence-files/'):
+                sequence_file_object = system.get('request').embed(file, '@@object?skip_calculated=true')
+                if not sequence_file_object.get('seqspec'):
+                    no_seqspec.append(file)
+        if no_seqspec:
+            no_seqspec = ', '.join([audit_link(path_to_text(file_no_seqspec), file_no_seqspec)
+                                   for file_no_seqspec in no_seqspec])
+            detail = (
+                f'File set {audit_link(path_to_text(value["@id"]), value["@id"])} has sequence file(s): '
+                f'{no_seqspec} which do not have a seqspec configuration file.'
+            )
+            yield AuditFailure('missing sequence specification', detail, level='ERROR')
+
+
+@audit_checker('FileSet', frame='object')
+def audit_missing_seqspec_files(value, system):
```

Comment:
i think the function should be renamed to reflect the audit it is executing. oyu are trying to detect files that are linked to other files that are expected to reside within the file set. So you are auditing for related files that are not in the file_set. audit_files_associated _with_incorrect_fileset or something along these lines